src/compiler/checker.ts

```diff
@@ -44446,77 +44452,125 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
-                            return left ** right;
+                            return { value: left ** right };
+                    }
+                }
+                else if (typeof left === "bigint" && typeof right === "bigint") {
+                    switch ((expr as BinaryExpression).operatorToken.kind) {
+                        case SyntaxKind.BarToken:
+                            return { value: left | right };
+                        case SyntaxKind.AmpersandToken:
+                            return { value: left & right };
+                        case SyntaxKind.GreaterThanGreaterThanToken:
+                            return { value: left >> right };
+                        case SyntaxKind.LessThanLessThanToken:
+                            return { value: left << right };
+                        case SyntaxKind.CaretToken:
+                            return { value: left ^ right };
+                        case SyntaxKind.AsteriskToken:
+                            return { value: left * right };
+                        case SyntaxKind.SlashToken:
+                            return { value: left / right };
+                        case SyntaxKind.PlusToken:
+                            return { value: left + right };
+                        case SyntaxKind.MinusToken:
+                            return { value: left - right };
+                        case SyntaxKind.PercentToken:
+                            return { value: left % right };
+                        case SyntaxKind.AsteriskAsteriskToken:
+                            return { value: left ** right };
                     }
                 }
                 else if (
-                    (typeof left === "string" || typeof left === "number") &&
-                    (typeof right === "string" || typeof right === "number") &&
+                    (typeof left === "string" || typeof right === "string") &&
                     (expr as BinaryExpression).operatorToken.kind === SyntaxKind.PlusToken
                 ) {
-                    return "" + left + right;
+                    return { value: "" + left + right };
                 }
                 break;
+            }
             case SyntaxKind.StringLiteral:
             case SyntaxKind.NoSubstitutionTemplateLiteral:
-                return (expr as StringLiteralLike).text;
-            case SyntaxKind.TemplateExpression:
-                return evaluateTemplateExpression(expr as TemplateExpression, location);
+                return { value: (expr as StringLiteralLike).text };
+            case SyntaxKind.TemplateExpression: {
+                const value = evaluateTemplateExpression(expr as TemplateExpression, location);
+                if (typeof value === "undefined") {
+                    break;
+                }
+                return { value };
+            }
             case SyntaxKind.NumericLiteral:
-                checkGrammarNumericLiteral(expr as NumericLiteral);

```

Comment:
I believe this was redundant here since it should already be checked elsewhere and it's not exactly the responsibility of this evaluation to raise errors like this.

tangent: I can't see any test case for this diagnostic
> Numeric literals with absolute values equal to 2^53 or greater are too large to be represented accurately as integers.(80008)